src/pages/admin/tokens/edit.tsx

```diff
@@ -96,6 +106,102 @@ interface MarketplaceForm {
   token: string
 }
 
+async function createAuctionHouses(
+  connection: Connection,
+  wallet: Wallet,
+  tokens: { address: string }[],
+  sellerFeeBasisPoints: number
+): Promise<{ address: string }[]> {
+  const publicKey = wallet.publicKey
+
+  const auctionHouses: { address: string }[] = []
+  const instructions: TransactionInstruction[] = []
+
+  tokens.forEach(async (token) => {
+    const canChangeSalePrice = false
+    const requiresSignOff = false
+    const treasuryWithdrawalDestination = undefined
+    const feeWithdrawalDestination = undefined
+    const treasuryMint = token.address
+
+    const twdKey = treasuryWithdrawalDestination
+      ? new PublicKey(treasuryWithdrawalDestination)
+      : wallet.publicKey
+
+    const fwdKey = feeWithdrawalDestination
+      ? new PublicKey(feeWithdrawalDestination)
+      : wallet.publicKey
+
+    const tMintKey = treasuryMint ? new PublicKey(treasuryMint) : NATIVE_MINT
+
+    const twdAta = tMintKey.equals(NATIVE_MINT)
+      ? twdKey
+      : (
+          await AuctionHouseProgram.findAssociatedTokenAccountAddress(
+            tMintKey,
+            twdKey
+          )
+        )[0]
+
+    const [auctionHouse, bump] =
+      await AuctionHouseProgram.findAuctionHouseAddress(
+        wallet.publicKey,
+        tMintKey
+      )
+
+    auctionHouses.push({ address: auctionHouse.toBase58() })
+
+    const [feeAccount, feePayerBump] =
+      await AuctionHouseProgram.findAuctionHouseFeeAddress(auctionHouse)
+
+    const [treasuryAccount, treasuryBump] =
+      await AuctionHouseProgram.findAuctionHouseTreasuryAddress(auctionHouse)
+
+    const auctionHouseCreateInstruction = createCreateAuctionHouseInstruction(
+      {
+        treasuryMint: tMintKey,
+        payer: wallet.publicKey,
+        authority: wallet.publicKey,
+        feeWithdrawalDestination: fwdKey,
+        treasuryWithdrawalDestination: twdAta,
+        treasuryWithdrawalDestinationOwner: twdKey,
+        auctionHouse,
+        auctionHouseFeeAccount: feeAccount,
+        auctionHouseTreasury: treasuryAccount,
+      },
+      {
+        bump,
+        feePayerBump,
+        treasuryBump,
+        sellerFeeBasisPoints,
+        requiresSignOff,
+        canChangeSalePrice,
+      }
+    )
+    instructions.push(auctionHouseCreateInstruction)
+  })
+
+  const transaction = new Transaction()
+
+  instructions.forEach((instruction: TransactionInstruction) => {
+    transaction.add(instruction)
+  })
+
+  transaction.feePayer = publicKey
+  transaction.recentBlockhash = (
+    await connection.getRecentBlockhash()
+  ).blockhash
+
+  const signedTransaction = await wallet.signTransaction!(transaction)
+  const txtId = await connection.sendRawTransaction(
+    signedTransaction.serialize()
+  )
+
+  if (txtId) await connection.confirmTransaction(txtId, 'confirmed')
+
+  return auctionHouses
+}
+
```

Comment:
This needs some work. Right now we are requiring the user to sign 2 transactions though this can likely all go in 1. 

I think this is work having in an instructions directory within the sdk package that setups and returns instructions for the caller to join together in a transaction. The marketplace edit action should have its instructions accessible from the instructions module so 1 can transaction can be built.